### PR TITLE
Add examples for custom graph in the doc of `soft_random_geometric_graph` and `thresholded_random_geometric_graph`

### PR DESCRIPTION
**Pull request by yuanx749**
Refer to https://github.com/networkx/networkx/pull/7730#discussion_r1857657982.

This PR added example code following the corresponding instructions of Custom Graph in the docstrings. Also correct the link of `scipy.stats`, which directed to 404 not found before.

### CODE REVIEW

File: networkx/generators/geometric.py [hunk 1]
```diff
@@ -281,6 +281,11 @@ def soft_random_geometric_graph(
     exponential distribution with rate parameter :math:`\lambda=1` if their
     Euclidean distance is at most 0.2.
 
+    >>> from scipy.stats import expon
+    >>> n = 100
+    >>> p_dist = lambda x: expon.pdf(x, scale=1)
+    >>> G = nx.soft_random_geometric_graph(n, 0.2, p_dist=p_dist)
+
```
Comment (Schefflera-Arboricola) [inline networkx/generators/geometric.py:288]: Thanks for adding this @yuanx749 ! If possible, please consider moving the example(explaining the usage of `pos` kwarg) from the `Notes` section to this `Example` section. And the whole `Example` section at the end(https://numpydoc.readthedocs.io/en/latest/format.html#examples). Thanks!

File: networkx/generators/geometric.py [hunk 2]
```diff
@@ -796,6 +801,11 @@ def thresholded_random_geometric_graph(
     a exponential distribution with rate = 5 are >= theta = 0.1 and their
     Euclidean distance is at most 0.2.
 
+    >>> import random
+    >>> n = 50
+    >>> weight = {i: random.expovariate(5) for i in range(n)}
+    >>> G = nx.thresholded_random_geometric_graph(n, 0.2, 0.1, weight=weight)
+
```
Comment (Schefflera-Arboricola) [inline networkx/generators/geometric.py:808]: same here.

File: networkx/generators/geometric.py (hunk 1 above)
Comment (yuanx749) [reply networkx/generators/geometric.py]: Thanks for your review and addressing the CI failure. @Schefflera-Arboricola . I've made the move, with wording slightly adjusted.

**Review by rossbar — APPROVED** [review]
I took the liberty of replacing the scipy reference (i.e. `.. [2]`) with intersphinx links, other than that LGTM, thanks @yuanx749 !

**Review by Schefflera-Arboricola — APPROVED** [review]

**Pull request merged**